parcels/field.py

```diff
@@ -967,15 +968,10 @@ def __init__(self, name, U, V, W=None):
 
     def dist(self, lon1, lon2, lat1, lat2, mesh):
         if mesh == 'spherical':
-            r = 360*60*1852/2/np.pi
             rad = np.pi/180.
-            x1 = r*np.cos(rad*lon1) * np.cos(rad*lat1)
-            y1 = r*np.sin(rad*lon1) * np.cos(rad*lat1)
-            z1 = r*np.sin(rad*lat1)
-            x2 = r*np.cos(rad*lon2) * np.cos(rad*lat2)
-            y2 = r*np.sin(rad*lon2) * np.cos(rad*lat2)
-            z2 = r*np.sin(rad*lat2)
-            return np.sqrt((x2-x1)**2 + (y2-y1)**2 + (z2-z1)**2)
+            deg2m = 1852 * 60.
+            lat = (lat1+lat2)/2.
+            return np.sqrt(((lon2-lon1)*deg2m*cos(rad * lat))**2 + ((lat2-lat1)*deg2m)**2)
```

Comment:
Also, use `np.cos` instead to normal `cos`?

Reply:
Oups, this is wrong (bad copy paste from the .h version). Flake8 should have found that...